src/commands/base.ts

```diff
@@ -228,4 +229,60 @@ export default abstract class Base extends Command {
     hasToken(): boolean {
         return this.authToken !== ''
     }
+
+    public async populateParameters<ResourceType>(
```

Comment:
is there any reason to even have these functions on the base command and not a util function? I was thinking of ripping out these methods since they don't really seem to use state from the Command, and it makes it harder to reuse these outside of the Command classes

Reply:
this method is generic enough that it can be used by most commands that require multiple argument/flags. It does use `authKey` and `projectKey` but yeah, I can see your point about it not accessing many of the base class properties directly. This wasn't really intended to be used outside of the Command classes - where are you trying to use it?

Reply:
actually i could make `prompts` a member of the base class and then reference it in the methods like `this.prompts` - which would be nice because then i can drop another param from the method. I wish i could also do the same with `flags` but `flags` is static.

Reply:
`filterPrompts` and `mergeFlagsAndAnswers` could def be util classes

Reply:
`populateParameters` I feel like is doing too much and it makes it hard not to duplicate code that aren't in Commands (e.g. in the ListOption class). Just something to think about, I might tackle it in my PRs